Docs for the rafted status check procedure.

tselmegbaasan opened the pull request:
As part of the 2DC support, we needed to make the rafted status check procedure public. And therefore, we also need to add a public documentation. I've kept it quite simple. But it may be too technical in some parts, let me know.

P.S It's required to be part of the 5.24 release.

neo-technology-commit-status-publisher commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1823.surge.sh

New pages:
- [undefined](https://neo4j-docs-operations-1823.surge.sh/operations-manual/5/clustering/monitoring/status-check)
<!-- Sticky Pull Request Commentdocs-pr-changes -->

nick-giles-neo commented on modules/ROOT/pages/clustering/monitoring/status-check.adoc:
```diff
@@ -0,0 +1,46 @@
+* *requester:* is `TRUE` for the server on which the procedure is run, and `FALSE` on the remaining servers.
+* *error:* contains the error message if there is one. An example of an error is that one of more of the requested databases doesn't exist on the requester.
+
+In general the `replicationSuccessful` field can be used to determine overall write-availability, whereas the `memberStatus` field can be checked in order to see whether the database is fault-tolerant or not.
```
Would it be worth explaining how you can determine 'not fault tolerant' from the statuses? E.g. reference the quorum calculation, and say `REPLICATING` is enough for data safety but `APPLYING` is needed if you want queries with bookmarks to work in a timely manner.

nick-giles-neo replied:
Nice

priyolahiri commented:
Would customers be familiar with the term "rafted" and what it actually means?

tselmegbaasan commented:
@priyolahiri That's a good question. I'll ask the team for input. I definitely think that removing the reference to raft can simplify for the user if they only intend to use the status check for checking availability. But there's also the leader information, and we might extend the procedure with other raft information in the future. So the common denominator for the displayed information is raft... But it might be too detailed...

neo-technology-commit-status-publisher commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1823.surge.sh

New pages:
- [undefined](https://neo4j-docs-operations-1823.surge.sh/operations-manual/5/clustering/monitoring/status-check)
<!-- Sticky Pull Request Commentdocs-pr-changes -->

Frannie-Ludmilla commented on modules/ROOT/pages/clustering/monitoring/status-check.adoc:
```diff
@@ -0,0 +1,57 @@
+* *recognisedLeader:* shows the server id of the perceived leader of each raft group member.
+* *recognisedLeaderTerm:* shows the term of the perceived leader of each raft group member. If the raft group members report different leaders, the one with the highest term should be trusted.
+* *requester:* is `TRUE` for the server on which the procedure is run, and `FALSE` on the remaining servers.
+* *error:* contains the error message if there is one. An example of an error is that one of more of the requested databases doesn't exist on the requester.
```
```suggestion
* *error:* contains the error message if there is one. An example of an error is that one or more of the requested databases doesn't exist on the requester.
```

Frannie-Ludmilla reviewed (commented):
In the clustering docs, there is no reference to Raft 😮 Not even in the glossary. I think that since it's important to mention it in this section, we might as well add a Trello Card for filling this gap.

NataliaIvakina reviewed (commented):
@tselmegbaasan, hi! Thank you for the page. I think this information will be super useful. However, I have several concerns. First, because of using completely new terms such as 'rafted check' and 'rafted databases'. Even 'raft group' is not very familiar to users. I think we mention the Raft protocol only in the page https://neo4j.com/docs/operations-manual/current/clustering/introduction/#clustering-primary-mode

tselmegbaasan commented on modules/ROOT/pages/clustering/monitoring/status-check.adoc:
```diff
@@ -0,0 +1,57 @@
+
+[NOTE]
+====
+The member on which the procedure is called replicates a `status check entry` in the same raft group as the transactions, and verifies that the entry can be replicated and applied.
+
+Since the entry is not applied to the transaction state machine, it's not guaranteed that the database is write available even though the status check reports that
```
You're right, I've removed the confusing details and made it much more vague. Just said "there can be other issues with the database" basically

Frannie-Ludmilla commented on modules/ROOT/pages/clustering/monitoring/status-check.adoc:
```diff
@@ -0,0 +1,57 @@
+[role=label--new-5.24]
+== Rafted Status Check
+
+Neo4j 5.24 introduces the xref:reference/procedures.adoc#procedure_dbms_cluster_statusCheck[`dbms.cluster.statusCheck()`] procedure, which can be used to monitor the ability to replicate in rafted databases, which in most cases means being able to write to the database. It can also
```
Hi @NataliaIvakina! 
I have some questions related to the newlines/ carriage breaks for the docs.
Do we have to use them only when we want to split a paragraph and use the soft-wrap/word wrap in our IDE if the line is really long? 
What about multiple newlines? Do multiple newlines get treated as one blank line as in Markdown or do we have to be careful with the amount of newlines as well?
Thanks a lot! 🤗

neo-technology-commit-status-publisher commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1823.surge.sh

New pages:
- [undefined](https://neo4j-docs-operations-1823.surge.sh/operations-manual/5/clustering/monitoring/status-check)
<!-- Sticky Pull Request Commentdocs-pr-changes -->

tselmegbaasan commented:
@priyolahiri @NataliaIvakina @Frannie-Ludmilla @nick-giles-neo Thanks all for the review. Since almost all of you pointed out that the term "raft" is too novel for the user, I've changed all occurrences of "raft" to "cluster". Also, I don't refer to entries either, I've simplified it to "transactions" because that's probably the only thing the user cares about.

neo-technology-commit-status-publisher commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1823.surge.sh

New pages:
- [Cluster status check](https://neo4j-docs-operations-1823.surge.sh/operations-manual/5/clustering/monitoring/status-check)
<!-- Sticky Pull Request Commentdocs-pr-changes -->

priyolahiri commented:
LGTM

NataliaIvakina reviewed (approved):
@tselmegbaasan, hi! I think everything looks good. I suggested some editorial and formatting related changes. Please, take a look

neo-technology-commit-status-publisher commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1823.surge.sh

New pages:
- [Cluster status check](https://neo4j-docs-operations-1823.surge.sh/operations-manual/5/clustering/monitoring/status-check)
<!-- Sticky Pull Request Commentdocs-pr-changes -->

neo-technology-commit-status-publisher commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1823.surge.sh

New pages:
- [Cluster status check](https://neo4j-docs-operations-1823.surge.sh/operations-manual/5/clustering/monitoring/status-check)
<!-- Sticky Pull Request Commentdocs-pr-changes -->

neo-technology-commit-status-publisher commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1823.surge.sh

New pages:
- [Cluster status check](https://neo4j-docs-operations-1823.surge.sh/operations-manual/5/clustering/monitoring/status-check)
<!-- Sticky Pull Request Commentdocs-pr-changes -->

neo-technology-commit-status-publisher commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1823.surge.sh

New pages:
- [Cluster status check](https://neo4j-docs-operations-1823.surge.sh/operations-manual/5/clustering/monitoring/status-check)
<!-- Sticky Pull Request Commentdocs-pr-changes -->

neo-technology-commit-status-publisher commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-1823.surge.sh

New pages:
- [Cluster status check](https://neo4j-docs-operations-1823.surge.sh/operations-manual/5/clustering/monitoring/status-check)
<!-- Sticky Pull Request Commentdocs-pr-changes -->
